Models/Weapons/Factories/Adapter/MineAdapter.cs

```diff
@@ -0,0 +1,25 @@
+﻿using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Text;
+using System.Threading.Tasks;
+using TheBattleShipClient.Services;
+using static TheBattleShipClient.Services.WeaponsService;
+
+namespace TheBattleShipClient.Models.Weapons.Factories.Adapter
+{
+    class MineAdapter : WeaponFactory
+    {
+        public MineAdapter(string token, string roomId)
+           : base(token, roomId)
+        {
+        }
+        public override async Task<Weapon> CreateWeapon(int x, int y)
+        {
+            MineFactory adaptee = new MineFactory();

```

Comment:
Pagal pattern'ą turėtų būti ne local kintamasis mineFactory, o private field - dabar gaunasi, jog mineFactory kuriamas kiekvieną kartą iš naujo. Be to kuriant mineFactory būtų galima jam perduoti token ir room id - nereikėtų perdavinėt kaskart.